homeassistant/helpers/__init__.py

```diff
@@ -98,14 +98,17 @@ def config_per_platform(config, domain, logger):
 
     while config_key in config:
         platform_config = config[config_key]
+        if type(platform_config) is not list:
```

Comment:
`type` does not verify inheritance, use `isinstance` instead: `isinstance(platform_config, str)`